libraries/abstractions/secure_sockets/test/iot_test_udp.c

```diff
@@ -498,6 +506,91 @@ static BaseType_t prvCheckRxTxBuffers( uint8_t * pucTxBuffer,
 }
 /*-----------------------------------------------------------*/
 
+/*
+ * @brief Test SendTo and RecvFrom echo server. Return pdFAIL if one of the operation fails, allowing retry.
+ * We expect the socket to not fail during retry.
+ */
+static BaseType_t prvSendRecvEchoTestHelper( Socket_t xSocket,
+                                             size_t xLength,
+                                             SocketsSockaddr_t * pxEchoServerAddress,
+                                             uint32_t ulTxCount )
+{
+    BaseType_t xResult;
+    BaseType_t xIntermResult;
+    uint8_t * pucTxBuffer = ( uint8_t * ) pcTxBuffer;
+    uint8_t * pucRxBuffer = ( uint8_t * ) pcRxBuffer;
+    SocketsSockaddr_t xReceiveAddress = { 0 };
+    uint32_t ulRetry;
+
+    xIntermResult = pdPASS;
+    xResult = pdPASS;
```

Comment:
Good point, the total number of tries should be udptestMAX_RETRY+1 since retry implies it had done at least one loop already. Will be consistent with prvSOCKETS_Threadsafe_SameSocketDifferentTasks().